core/templates/dev/head/pages/exploration_player/ExplorationEngineService.js

```diff
@@ -241,74 +274,31 @@ oppia.factory('ExplorationEngineService', [
       getExplorationVersion: function() {
         return version;
       },
-      getStateContentHtml: function() {
-        return exploration.getUninterpolatedContentHtml(currentStateName);
-      },
-      getStateContentAudioTranslations: function() {
-        return exploration.getAudioTranslations(currentStateName);
-      },
       getStateContentAudioTranslation: function(stateName, languageCode) {
         return exploration.getAudioTranslation(stateName, languageCode);
       },
-      isContentAudioTranslationAvailable: function() {
-        return Object.keys(
-          exploration.getAudioTranslations(currentStateName)).length > 0 ||
-          AudioTranslationLanguageService.isAutogeneratedAudioAllowed();
-      },
-      getCurrentInteractionHtml: function(labelForFocusTarget) {
+      doesInteractionSupportHints: function() {
         var interactionId = exploration.getInteractionId(currentStateName);
-        if (!interactionId) {
-          return null;
-        }
-
-        return ExplorationHtmlFormatterService.getInteractionHtml(
-          interactionId,
-          exploration.getInteractionCustomizationArgs(currentStateName),
-          true,
-          labelForFocusTarget);
-      },
-      getNextInteractionHtml: function(labelForFocusTarget) {
-        var interactionId = exploration.getInteractionId(nextStateName);
-
-        return ExplorationHtmlFormatterService.getInteractionHtml(
-          interactionId,
-          exploration.getInteractionCustomizationArgs(nextStateName),
-          true,
-          labelForFocusTarget);
-      },
-      getNextInteraction: function() {
-        return exploration.getInteraction(nextStateName);
+        return (
+          !INTERACTION_SPECS[interactionId].is_terminal &&
+        !INTERACTION_SPECS[interactionId].is_linear);
       },
       getCurrentInteraction: function() {
```

Comment:
Why do we need all these individual functions still? I looked up a couple of them and IMO we should be able to just get them from nextCard, right?

Reply:
Yeah, this was used in HintAndSolutionModalService, where a card object is not present. Should I take card object from transcript there also?

Reply:
Can you pass the relevant details from the card object into the modal?

Reply:
Done. Hints and solution directive is called from exploration_footer, so no way to pass card object directly from conversation skin, but there was already a broadcast that initialised hints in it. So, modified that to send card object and initialise hints and solution using that.